### ISSUE TITLE
Ability to execute dataset queries

### ISSUE THREAD

**Issue by lbendlin** [migueesc123/PowerBIRESTAPI#101]
In addition to PowerBIRESTAPI.GETData  it would be nice to have a PowerBIRESTAPI.POSTData function that would accept an additional Content parameter. That way we could run queries that do things like datasets/xxx/executequeries .

**Comment by migueesc123** [comment]
Power Query wouldn't be the best tool for such scenarios. Primarily because there's no guarantee that the query will be executed only once. It could have some retry operations, it could be evaluated during a data preview for cache or you could be working with a cached version of the data. Something like an app created over PowerApps or perhaps a button created with Power Automate would be the best choice for such POST scenarios.

**Comment by lbendlin** [comment]
Fair enough. I implemented it in my copy of the connector, and it seems to work fine.  The real reason for pushing for this is that we need to use this feature on an enterprise gateway and AAD auth is our only option for the moment.

**Comment by migueesc123** [comment]
> Fair enough. I implemented it in my copy of the connector, and it seems to work fine. The real reason for pushing for this is that we need to use this feature on an enterprise gateway and AAD auth is our only option for the moment.

Apart from what I mentioned before, It can def work but ,the implementation could be cumbersome and might not be exactly what people are expecting.

I'm hoping that at one point the MSFT team behind the Power BI REST API will come up with their own custom connector that will completely replace this project.

However, as you mentioned, you can absolutely modify the connector to your specific needs.
